nupic/research/frameworks/dynamic_sparse/networks/layers.py

```diff
@@ -164,7 +180,15 @@ def reset_coactivations(self):
     def _update_coactivations(self, input_tensor, output_tensor):
 
         new_coacts = self.calc_coactivations(input_tensor, output_tensor)
-        self.coactivations[:] += new_coacts
+        if self.update_func:
+            updated_coacts = self.update_func(self.coactivations, new_coacts)
+        elif self.moving_average_alpha:
+            alpha = self.moving_average_alpha
+            updated_coacts = (1 - alpha) * self.coactivations + alpha * new_coacts
+        else:
+            updated_coacts = self.coactivations + new_coacts
+
+        self.coactivations[:] = updated_coacts
```

Comment:
In the case of moving average, should we still reset it after each epoch? Maybe we can take that off in the case of moving average and move from the concept of "epoch" to just iterations.